backend/src/apiserver/resource/resource_manager.go

```diff
@@ -495,7 +498,7 @@ func (r *ResourceManager) CreateRun(ctx context.Context, run *model.Run) (*model
 		RunId: run.UUID,
 		RunAt: run.RunDetails.CreatedAtInSec,
 	}
-	executionSpec, err := tmpl.RunWorkflow(run, runWorkflowOptions)
+	executionSpec, err := tmpl.RunWorkflow(run, runWorkflowOptions, r.proxyConfig)
```

Comment:
On second thought, it might be beneficial to have it be part of the `Template` interface/objects such as adding a method called `GetProxyConfig` or something so that way the proxy configuration is stored on the template and doesn't need to be passed around.

Reply:
It will increase coupling. `Template` [is created in `PipelineVersionsWebhook`](https://github.com/kubeflow/pipelines/blob/2efcde5efd3952b91ea79a5ee6dbf064282f719a/backend/src/apiserver/webhook/pipelineversion_webhook.go#L86C20-L86C33), for example. `PipelineVersionsWebhook` doesn't need to know `proxy.Config`, but it would need to depend on it to create a `Template`.
Unless we make it optional and lose the benefit of having the compiler on our side 🤷🏽‍♂️

Reply:
No longer applies, since we changed how to interact with `proxy.Config`.